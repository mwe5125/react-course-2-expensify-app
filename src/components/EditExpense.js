import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { editExpense, removeExpense } from '../actions/expenses'

// refactor edit expense page to be a class based component
// setup mapDispatchToProps...editExpense and removeExpense

// should render EditExpensePage
  //snapshot

// should hanlde editExpense
  //spies

// should handle removeExpense
    // just setup similar to first one of last

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense)
        this.props.history.push('/')
    }
    onClick = () => {
        this.props.removeExpense({id: this.props.expense.id})
        this.props.history.push('/')
    }
    render () {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense} 
                    onSubmit={this.onSubmit}
                />
                <button 
                    onClick={this.onClick}
                    >
                Remove
                </button>
            </div>
            )
        }
    }


const mapDispatchToProps = (dispatch) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (data) => dispatch(removeExpense(data))
})

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)