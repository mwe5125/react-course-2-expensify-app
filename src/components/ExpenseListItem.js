import React from 'react'
import { Link } from 'react-router-dom'

// Export a stateless functional componenet
// description, amount, createdAt

export const ExpenseListItem = ({ description, amount, createdAt, id }) => (
        <div>
            <Link to={`/edit/${id}`}>
                <h3>{description}</h3>
            </Link>
            <p>{amount} - {createdAt}</p>
        </div>
    )


export default ExpenseListItem
