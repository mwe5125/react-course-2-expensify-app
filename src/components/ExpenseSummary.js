import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import getExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    )
}

// how I did it which worked
// const ExpenseSummary = (props) => (
//     <div>
//         {   
//             props.expenses.length >= 1 ? (
//                 <p>
//                     Viewing {props.expenses.length} {props.expenses.length > 1 ? 'expenses' : 'expense'} totalling {numeral((getExpensesTotal(props.expenses)/100)).format('$0,0.00')}
//                 </p>
//             ) :  (<span></span>)
//         }
//     </div>
// )



const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    }
}


export default connect(mapStateToProps) (ExpenseSummary)