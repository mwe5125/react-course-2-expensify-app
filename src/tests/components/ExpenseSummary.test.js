import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'

test('should render Expense summary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render Expense summary with multiple expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={5} expensesTotal={2356781} />)
    expect(wrapper).toMatchSnapshot()
})