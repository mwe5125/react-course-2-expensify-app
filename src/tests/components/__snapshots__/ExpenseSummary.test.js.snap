// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should render Expense summary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    5
     
    expenses
     totalling 
    $23,567.81
  </h1>
</div>
`;
